marklogic-data-hub/src/test/java/com/marklogic/hub/step/impl/IngestWithSourceNameAndSourceTypeTest.java

```diff
@@ -0,0 +1,89 @@
+package com.marklogic.hub.step.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.hub.AbstractHubCoreTest;
+import com.marklogic.hub.dataservices.FlowService;
+import com.marklogic.hub.dataservices.StepService;
+import com.marklogic.hub.flow.FlowInputs;
+import com.marklogic.hub.flow.FlowRunner;
+import com.marklogic.hub.flow.RunFlowResponse;
+import com.marklogic.hub.flow.impl.FlowRunnerImpl;
+import com.marklogic.hub.job.JobStatus;
+import com.marklogic.hub.test.ReferenceModelProject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class IngestWithSourceNameAndSourceTypeTest extends AbstractHubCoreTest {
+
+    ReferenceModelProject project;
+
+    @BeforeEach
+    void beforeEach() {
+        project = installReferenceModelProject();
+    }
+
+    @Test
+    void runDefaultIngestionWithSourceNameAndSourceType(){
+        String path = "entity-reference-model/data/customers";
+        ObjectNode info = new ObjectMapper().createObjectNode();
+        info.put("name", "sources-test");
+        info.put("description", "step to test source name and source type");
+        info.put("sourceFormat", "json");
+        info.put("targetFormat", "json");
+        info.put("sourceName", "sample-source");
+        info.put("sourceType", "employee");
+        info.put("outputURIPrefix", "");
+
+        StepService.on(getHubClient().getStagingClient()).saveStep("ingestion", info, false);
+
+        // Add it to the existing echoFlow, which is assumed to have one step
+        FlowService.on(getHubClient().getStagingClient()).addStepToFlow("echoFlow", "sources-test", "ingestion");
+
+        // And run the flow
+        FlowInputs inputs = new FlowInputs("echoFlow", "2");
+        inputs.setInputFilePath(getClass().getClassLoader().getResource(path).getPath());
+
+        FlowRunner flowRunner = new FlowRunnerImpl(getHubClient());
+        RunFlowResponse response = flowRunner.runFlow(inputs);
+        flowRunner.awaitCompletion();
+        assertEquals(JobStatus.FINISHED.toString(), response.getJobStatus());
+        JsonNode sourceDoc = getHubClient().getStagingClient().newJSONDocumentManager().read("customer1.json").next().getContent(new JacksonHandle()).get();
```

Comment:
Can use getStagingDoc instead - less lines of code is almost always good in a test, as it makes it easier to read and understand.